### DISCUSSION TITLE
How do I get structural equality while using the HashSet ?

### DISCUSSION THREAD

**Discussion by jayaprakashanand** [js-sdsl/js-sdsl#190] [Q&A]
I need to find if a class object with same values (toString method) already exists in the collection. This does not work in the HashSet but works in the ordered set. Am I missing something?

**Comment by ZLY201** [comment]
Try this:
```ts
function isOrderedSetEqual(st1: OrderedSet<unkonwn>, st2: OrderedSet<unkonwn>) {
    if (st1.size() !== st2.size()) {
        return false;
    }
    return Array.from(st1).toString() === Array.from(st2).toString();
}
```

**Reply by jayaprakashanand** [reply]
Iimport { HashSet } from "js-sdsl";


class Vehicle{
    /**
     *
     */
    constructor(public name: string, public model: string, public owner: string) {
        
        
    }
  

   }


const testHashSet =() =>{
    let  tset = new HashSet<Vehicle>();

    let x = new Vehicle("Honda", "crv","anand");
    let y = new Vehicle("nissan","versa","sown");
    let z = new Vehicle("Honda", "crv","anand");

    tset.insert(x);
    tset.insert(y);


    let it = tset.find(z);  // this should return a iterator since x and z have equal values... How do i achieve this ????
    if (it.equals(tset.end())){
        tset.insert(z);
    }else{
        console.log(it.pointer)
        console.log('already found');
    }
   
    console.log(tset.length);
}

testHashSet();

**Comment by ZLY201** [comment]
ah sorry i miss the question at the first time, i'm afraid the `HashSet` doesn't support this function now, to check if two elements is equal when using hash container you can do things like this:
```ts
class Vehicle {
    constructor(name, model, owner) {
        this.name = name;
        this.model = model;
        this.owner = owner;
    }
    // your hash function here, please ensure it returns a unique value for whole database
    toString() { return this.name + this.model + this.owner; }
}

const x = new Vehicle("Honda", "crv", "anand");
const y = new Vehicle("nissan", "versa", "sown");
const z = new Vehicle("Honda", "crv", "anand");

const mp = new HashMap();

mp.setElement(x.toString(), x);
mp.setElement(y.toString(), y);

// Then it works well

let it = mp.find(z.toString());
if (it.equals(mp.end())) {
    mp.setElement(z.toString(), z);
} else {
    console.log(it.pointer)
    console.log('already found');
}

console.log(mp.length);    // logs 2 because z is equals to x
```
